EventFilter/EcalRawToDigi/plugins/UnpackGPU.cu

```diff
@@ -151,16 +180,48 @@ namespace ecal {
       // fed header
       //
       auto const fed_header = buffer[0];
-      uint32_t bx = (fed_header >> 20) & 0xfff;
-      uint32_t lv1 = (fed_header >> 32) & 0xffffff;
+      uint32_t bx = (fed_header >> H_BX_B) & H_BX_MASK;
+      uint32_t lv1 = (fed_header >> H_L1_B) & H_L1_MASK;
+      uint32_t triggerType = (fed_header >> H_TTYPE_B) & H_TTYPE_MASK;
+
+      // determine the number of FE channels from the trigger type
+      uint32_t numbChannels(0);
+      if (triggerType == PHYSICTRIGGER) {
+        numbChannels = NUMB_FE;
+      } else if (triggerType == CALIBRATIONTRIGGER) {
+        numbChannels = NUMB_FE + 2;  // FE + 2 MEM blocks
+      } else {
+        // unsupported trigger type
+        return;
+      }
```

Comment:
what about "random" triggers ?

Reply:
I assume the are counted as physics type as well.
This block is essentially the same than in the CPU unpacker https://github.com/cms-sw/cmssw/blob/master/EventFilter/EcalRawToDigi/src/DCCEBEventBlock.cc#L165-L177

The possible types are defined here: https://github.com/cms-sw/cmssw/blob/master/EventFilter/EcalRawToDigi/interface/DCCRawDataDefinitions.h#L15-L18 and it seems the later two are not considered.

Reply:
Mhm, where does the value of `triggerType` come from ?

Reply:
What I mean is, is it supposed to match the "trigger type" set by the TCDS, or does it use a different categorisation (sigh) ?

Reply:
It comes from the DCC block header.

From CMS NOTE 2005/021:
> DETAILED TRIGGER TYPE: A Timing and Trigger Control (TTC) “long command” [5] written in the DCC
header during calibration triggers expected at the orbit gaps.

Reply:
I do not think that it is supposed to match the TCDS one.

Reply:
I checked with 1k events of PhysicsZeroBias data from run 346512 that had >20kHz randoms and they had all trigger type 1 (PHYSICTRIGGER). But I am not sure if randoms make it into that dataset.

Reply:
OK, to be sure I checked on a CRAFT file from last weekend, pre-selecting "random" events with a dedicated filter
```python
process.randomEventsFilter = cms.EDFilter( "HLTTriggerTypeFilter",
    SelectedTriggerType = cms.int32( 3 )
)
```
and indeed the ECAL unpacker sees them as "PHYSICTRIGGER" type:
```
...
Begin processing the 52nd record. Run 349527, Event 575973, LumiSection 1 on stream 0 at 02-Apr-2022 10:23:19.893 CEST
Begin processing the 53rd record. Run 349527, Event 308265, LumiSection 1 on stream 0 at 02-Apr-2022 10:23:19.893 CEST
ECAL DCC FED 613, trigger type 1
ECAL DCC FED 653, trigger type 1
ECAL DCC FED 618, trigger type 1
ECAL DCC FED 638, trigger type 1
ECAL DCC FED 608, trigger type 1
...
```